### PR TITLE
Un-abstractify base class

### PR DESCRIPTION
**Pull request by zachmullen**
It now provides sensible default behaviors. Plugin entry points now no longer need to provide an `__init__` unless they actually need it. Additionally, plugins no longer strictly need to provide a task listing method, to support plugins that might not expose custom tasks.

### CODE REVIEW

File: girder_worker/__init__.py [hunk 1]
```diff
@@ -33,20 +31,17 @@
 logger = log_utils.setupLogger(config)
 
 
-@add_metaclass(abc.ABCMeta)
 class GirderWorkerPluginABC(object):
```
Comment (subdavis) [inline girder_worker/__init__.py:34]: Should this still be named `ABC` if it's not a true ABC?  Seems like just a regular base class now.

Comment (zachmullen) [reply girder_worker/__init__.py:34]: I don't think it should, however I'm not trying to make a breaking change that would be trouble for all downstream plugins. As written, this should be non-breaking.

**Comment by kotfic** [comment]
I'm fine with this,  though it does feel like a ripe opportunity for some primo bike-shedding =P

**Comment by zachmullen** [comment]
Could you approve @kotfic ?

**Pull request merged**